zstd/enc_best.go

```diff
@@ -197,12 +197,13 @@ encodeLoop:
 
 		// Set m to a match at offset if it looks like that will improve compression.
 		improve := func(m *match, offset int32, s int32, first uint32, rep int32) {
-			if s-offset >= e.maxMatchOff || load3232(src, offset) != first {
+			delta := s - offset
+			if delta >= e.maxMatchOff || delta <= 0 || load3232(src, offset) != first {
 				return
 			}
 			if debugAsserts {
-				if offset <= 0 {
-					panic(offset)
+				if offset >= s {
```

Comment:
That debug check can never be triggered (`offset >= s` <=> `0 >= offset -s` <=> `0 >= delta`) now that the previous `if [...] delta <= 0` has been added. Is that intentional?

Reply:
I don't mind debugAsserts being a bit "excessive" in its testing. It is removed from regular builds anyway, and mostly for debugging.